feat: add docs for searchbase (#156)
appbaseio/Docs: content/docs/reactivesearch/searchbase/overview/QuickStart.md

---
title: 'Quickstart'
meta_title: 'Quickstart to SearchBase'

> meta_description: 'SearchBase is a lightweight & platform agnostic library to build the search UIs with Elasticsearch.'

Reviewer: ```suggestion
meta_description: 'SearchBase is a lightweight and platform agnostic library that provides scaffolding to create search experiences powered by Elasticsearch.'
```

Reviewer: @bietkul Use consistent naming convention for Searchbase. Is it SearchBase, Searchbase or searchbase? I see use of all three in the doc PR here.

nestedSidebar: 'searchbase-reactivesearch'
---

> [searchbase](https://github.com/appbaseio/searchbase) - A lightweight & platform agnostic library to build the search UIs.

Reviewer: wrong link

Reviewer: Update description as per above.

To use the umd build, add the following script in your `index.html` file.

```js

> <script defer src="https://cdn.jsdelivr.net/npm/@appbaseio/searchbase@1.0.0/dist/@appbaseio/searchbase.umd.min.js"></script>

Reviewer: Is the version referenced correctly to match the newest release?

Author: @siddharthlatest I will create a stable release

<script defer src="https://cdn.jsdelivr.net/npm/@appbaseio/searchbase@1.0.0/dist/@appbaseio/searchbase.umd.min.js.map"></script>
```

## Basic Usage

### A simple example

The following example creates a single component of type `search` to render the results based on the input value.

```js

> import { Component } from '@appbaseio/searchbase';

Reviewer: renaming is remaining.

});

// Get the input element
const searchElement = document.getElementById('search');

// Bind the searchComponent value to input value
searchElement.value = searchComponent.value;

// Update the search input value to searchComponent to fetch the results
searchElement.addEventListener('input', e => {
	// To fetch the suggestions based on the value changes
	searchComponent.setValue(e.target.value);
});

// Build DOM when search results update
searchComponent.subscribeToStateChanges(
	change => {
		const results = change.results.next;
		const resultsElement = document.getElementById('results');
		resultsElement.innerHTML = '';
		results.data.forEach(element => {
			var node = document.createElement('LI'); // Create a <li> node
			var resultNode = document.createTextNode(element.name); // Create a text node
			node.appendChild(resultNode); // Append the text to <li>
			resultsElement.appendChild(node);
		});
	},
	['results'],
);

// Fetch the default results at initial load
searchComponent.triggerDefaultQuery();
```

Add this in your HTML

```html
<input placeholder="type to search" id="search" />

<div id="results"></div>
```

> 

Reviewer: Add a CSB link with the same example code in there so a user can visually see how this looks in the UI.

### An example with a facet

The following example creates three components:

-   a search component to perform the search,
-   a filter component to filter the GitHub repo by languages,
-   a result component to render the results based on the select language filters and search query

> The result component has a dependency on the search and language filter components (see the `react` property) i.e reactive to the changes for search and filter component in a way that whenever the search/filter value will change the result component will trigger an `Elasticsearch` query to update results based on the search/filter value.

Reviewer: ```suggestion
The result component watches for changes to the search and language filter components (see the `react` property). It reacts to the inputs and filter selection changesby triggering an `Elasticsearch` query to update results.
```

> The language filter component also depends on the search component which means that it will update the language options based on the search query. For example, if somebody searches for `angular` then the language filter will show `javascript` as an option.

Reviewer: See the above comment suggestion to change accordingly
